hawkmoth/__init__.py

```diff
@@ -42,13 +42,29 @@ class CAutoDocDirective(Directive):
     }
     has_content = False
 
+    def __display_parser_diagnostics(self, errors):
+        env = self.state.document.settings.env
+        for diag in errors:
+            severity, filename = diag[0], diag[1]
+            lineno, reason = diag[2], diag[3]
+            toprint = '{}:{}, {}'.format(filename, lineno, reason)
```

Comment:
This repeats the filename and line number. Real example:
`/home/bruno/Documents/projects/bms-contrib/hawkmoth/doc/examples.rst:262: WARNING: /home/bruno/Documents/projects/bms-contrib/hawkmoth/doc/examples/example-70-function.c:9, declaration of 'struct list' will not be visible outside of this function`

This must be `self.logger.warning(reason, location(env.docname, self.lineno))` and so on below. You can get rid of this variable.

Reply:
Wait, actually it doesn't repeat it, there's just 2 sources involved of course. Sorry about that, I think it's probably best to show both.

Reply:
Yep, I think it is better this way even though at first sight it looks strange.